core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2551,6 +2551,7 @@ private[util] object CallerContext extends Logging {
  */
 private[spark] class CallerContext(
    from: String,
+   upstreamCallerContext: Option[String] = None,
```

Comment:
I agree with you that new parameters should be added as last parameters so that no client code would be broken. It’s just in this case, there are three callers in total and they are all under my control. The new optional parameter would be used much more frequently than other optional parameters, so I think I shall change the parameter orders before the class is used more widely. If I put the new parameter as the last one, users (including the existing three) would have to pass may “None”s as parameters.